send_review_notification/github_services.py

```diff
@@ -0,0 +1,266 @@
+                        datetime.timedelta(hours=wait_hours)
+                    ):
+                    reviewer_to_assigned_prs[reviewer.name].append(pull_request)
+    return reviewer_to_assigned_prs
+
+
+# Here we use type Any because the response we get from the api call is hard
+# to annotate in a typedDict, Hence used type Any.
+def __process_activity(
+    pull_request: github_domain.PullRequest,
+    event: Dict[str, Any]
+) -> None:
+    """Process activity and updates assignee timestamps."""
+    if event['event'] != 'assigned':
+        return
+
+    assignee = pull_request.get_assignee(event['assignee']['login'])
+    event_timestamp = parser.parse(event['created_at'])
+    if assignee:
+        assignee.set_timestamp(max([assignee.timestamp, event_timestamp]))
+
+
+def update_assignee_timestamp(
+    org_name: str,
+    repository: str,
+    pr_list: List[github_domain.PullRequest]
+) -> None:
+    """Fetches PR timeline and updates assignment timestamp."""
+    for pull_request in pr_list:
+        pr_number = pull_request.number
+        activity_url = ISSUE_TIMELINE_URL_TEMPLATE.format(
+            org_name, repository, pr_number)
+
+        page_number = 1
+        while True:
+            logging.info('Fetching PR #%s timeline', pr_number)
+            response = requests.get(
+                activity_url,
+                params={'page': page_number, 'per_page': 100},
+                headers={
+                    'Accept': 'application/vnd.github.mockingbird-preview+json',
+                    'Authorization': f'token {_TOKEN}'}
+            )
+            response.raise_for_status()
+            timeline_subset = response.json()
+
+            if len(timeline_subset) == 0:
+                break
+
+            for event in timeline_subset:
+                __process_activity(pull_request, event)
+
+            page_number += 1
+
+
+@check_token
+def create_discussion_comment(org_name: str, repo: str, body: str) -> None:
+    """Creates github discussion on the team_slug with the given title and
+    body.
+    """
+
+    query_category_id = """
+        query($org_name: String!, $repository: String!) {
+            repository(owner: $org_name, name: $repository) {
+                discussionCategories(first: 10) {
+                    nodes {
+                        id
+                        name
+                    }
+                }
+            }
+        }
+    """
+
+    variables = {
+        'org_name': org_name,
+        'repository': repo
+    }
+
+    response = requests.post(
+        GITHUB_GRAPHQL_URL,
+        json={'query': query_category_id, 'variables': variables},
+        headers=_get_request_headers()
+    )
+
+    data = response.json()
+
+    category_id = ''
+    for category in data['data']['repository']['discussionCategories']['nodes']:
+        if category['name'] == 'Reviewer notifications':
+            category_id = category['id']
+            break
+
```

Comment:
Please add some error handling for the case where there is no `Reviewer notifications` category. I'd recommend throwing an error explaining that the category needs to be created

Reply:
I did. PTAL